libs/langchain/langchain/chains/router/llm_router.py

```diff
@@ -55,8 +55,10 @@ def _call(
         callbacks = _run_manager.get_child()
         output = cast(
             Dict[str, Any],
-            self.llm_chain.predict_and_parse(callbacks=callbacks, **inputs),
+            self.llm_chain.predict(callbacks=callbacks, **inputs),
         )
+        if self.llm_chain.prompt.output_parser is not None:
+            output = self.llm_chain.prompt.output_parser.parse(output)
```

Comment:
the goal of deprecating `predict_and_parse` is actually to avoid storing the output parser on the prompt. Instead, can you update to store on the `llm_chain` instead?